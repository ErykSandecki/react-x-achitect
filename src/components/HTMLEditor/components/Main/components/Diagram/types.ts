// types
import { BlockType, PortType, T2DCoordinates, TBlock, TConnector } from 'types';

export type TDraggableConnectorSource = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnectorTarget = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnector = {
  showIndicator: boolean;
  source?: TDraggableConnectorSource;
  sourceCoordinates: T2DCoordinates;
  target?: TDraggableConnectorTarget;
  targetCoordinates: T2DCoordinates;
};

export type TPossiblePort = T2DCoordinates & {
  blockId: string;
  blockType: BlockType;
  height: number;
  portType: PortType;
  width: number;
};

export type TShadowConnector = Pick<
  TConnector,
  'joinOperator' | 'relatedWithBlockId' | 'towardsTo'
> & {
  sourceBlock: Partial<TBlock> & { portType: PortType };
  targetBlock: Partial<TBlock> & { portType: PortType };
};
