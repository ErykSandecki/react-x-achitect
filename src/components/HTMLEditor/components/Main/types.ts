import { MutableRefObject } from 'react';

export type TMainRefs = {
  blocksPanelRef: MutableRefObject<HTMLDivElement>;
  diagramRef: MutableRefObject<HTMLDivElement>;
  panelPropertiesRef: MutableRefObject<HTMLDivElement>;
};
