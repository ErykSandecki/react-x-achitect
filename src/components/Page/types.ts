import { ReactNode } from 'react';

// types
import { PageType } from './enums';

export type TPageType = Record<PageType, string>;

export type TRestProps<T> = {
  children: ReactNode;
  pageType?: T;
  translationNameSpace?: string;
  withBackground?: boolean;
};
