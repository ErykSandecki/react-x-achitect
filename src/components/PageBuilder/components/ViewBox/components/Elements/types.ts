// types
import { MouseMode } from 'components/PageBuilder/enums';
import { TElement } from 'types';

export type TElementProps = {
  className: string;
  id: TElement['id'];
  mouseMode: MouseMode;
  parentId: TElement['parentId'];
  type: TElement['type'];
};
