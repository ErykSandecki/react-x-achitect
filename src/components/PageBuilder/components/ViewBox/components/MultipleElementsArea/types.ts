// types
import { TElement, TRectCoordinates } from 'types';

export type TElements = {
  elementsCordinates: Array<{
    coordinates: TRectCoordinates;
    id: TElement['id'];
  }>;
  outline: TRectCoordinates;
};
