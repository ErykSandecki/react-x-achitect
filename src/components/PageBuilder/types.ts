// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
