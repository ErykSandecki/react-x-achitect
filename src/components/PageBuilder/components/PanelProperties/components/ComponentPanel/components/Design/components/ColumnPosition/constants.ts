// others
import { translationNameSpace as parentNameSpace } from '../../constants';

export const MAX = 999999999;
export const MIN = -999999999;
export const translationNameSpace = `${parentNameSpace}.columnPosition`;
