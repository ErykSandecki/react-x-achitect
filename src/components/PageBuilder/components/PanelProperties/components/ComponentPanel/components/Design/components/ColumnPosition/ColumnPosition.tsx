import { FC, useRef } from 'react';
import { first, size } from 'lodash';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

// components
import { ScrubbableInput, Small, UITools } from 'shared';

// others
import { MAX, MIN, translationNameSpace } from './constants';

// hooks
import { usePositionEvents } from './hooks/usePositionEvents';

// store
import {
  dynamicDataSelector,
  elementAllDataSelectorCreator,
  selectedElementsSelector,
} from 'store/pageBuilder/selectors';
import { updateEventsStatus } from 'store/pageBuilder/actions';

// types
import { ColorsTheme, KeyboardKeys } from 'types';

// utils
import { handleSubmitInput } from 'utils';
import { isMixed } from './utils/isMixed';

const ColumnPosition: FC = () => {
  const dispatch = useDispatch();
  const dynamicData = useSelector(dynamicDataSelector);
  const selectedElements = useSelector(selectedElementsSelector);
  const firstElement = first(selectedElements);
  const element = useSelector(elementAllDataSelectorCreator(firstElement.id));
  const refInputX = useRef<HTMLInputElement>(null);
  const refInputY = useRef<HTMLInputElement>(null);
  const { t } = useTranslation();
  const isMultiple = size(selectedElements) > 1;
  const isMixedX = isMixed('x', dynamicData, firstElement, selectedElements);
  const isMixedY = isMixed('y', dynamicData, firstElement, selectedElements);
  const isRelative = selectedElements.some(
    ({ position }) => position === 'relative',
  );
  const { onBlurX, onBlurY, onChangeX, onChangeY, onMouseDown, x, y } =
    usePositionEvents(element, isMixedX, isMixedY, isMultiple, isRelative);

  return (
    <UITools.SectionColumn labels={[t(`${translationNameSpace}.label`)]}>
      <UITools.TextField
        disabled={isRelative}
        e2eValue="x"
        onBlur={onBlurX}
        onChange={(event) => onChangeX(event.target.value)}
        onClick={() => refInputX.current.select()}
        onKeyDown={(event) =>
          handleSubmitInput(KeyboardKeys.enter, refInputX.current)(event)
        }
        ref={refInputX}
        startAdornment={
          <ScrubbableInput
            disabled={isRelative}
            e2eValue="x"
            max={MAX}
            min={MIN}
            onChange={(value) => onChangeX(value.toString(), true)}
            onMouseDown={onMouseDown}
            onMouseUp={() =>
              dispatch(updateEventsStatus({ isMultipleMoving: false }))
            }
            value={isMultiple ? 0 : parseFloat(x)}
          >
            <Small color={ColorsTheme.neutral2}>X</Small>
          </ScrubbableInput>
        }
        type={isMixedX ? 'text' : 'number'}
        value={x}
      />
      <UITools.TextField
        disabled={isRelative}
        e2eValue="y"
        onBlur={onBlurY}
        onChange={(event) => onChangeY(event.target.value)}
        onClick={() => refInputY.current.select()}
        onKeyDown={(event) =>
          handleSubmitInput(KeyboardKeys.enter, refInputY.current)(event)
        }
        ref={refInputY}
        startAdornment={
          <ScrubbableInput
            disabled={isRelative}
            e2eValue="y"
            max={MAX}
            min={MIN}
            onChange={(value) => onChangeY(value.toString(), true)}
            onMouseDown={onMouseDown}
            onMouseUp={() =>
              dispatch(updateEventsStatus({ isMultipleMoving: false }))
            }
            value={isMultiple ? 0 : parseFloat(y)}
          >
            <Small color={ColorsTheme.neutral2}>Y</Small>
          </ScrubbableInput>
        }
        type={isMixedY ? 'text' : 'number'}
        value={y}
      />
    </UITools.SectionColumn>
  );
};

export default ColumnPosition;
