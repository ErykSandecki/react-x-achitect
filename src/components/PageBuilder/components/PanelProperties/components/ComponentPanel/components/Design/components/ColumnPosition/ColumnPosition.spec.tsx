import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import ColumnPosition from './ColumnPosition';

// mocks
import {
  elementAllDataMock,
  elementDynamicDataMock,
  elementStaticDataMock,
  pageBuilderStateMock,
  selectedElementMock,
} from 'test/mocks/reducer/pageBuilderMock';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { E2EAttribute } from 'types';

// utils
import { getByE2EAttribute } from 'test';

const currentPage = pageBuilderStateMock[PAGE_BUILDER].pages['0'];
const stateMock = {
  ...pageBuilderStateMock,
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...currentPage,
        elements: {
          allData: {
            ['-1']: {
              ...currentPage.elements.allData['-1'],
              children: [elementAllDataMock.id],
            },
            [elementAllDataMock.id]: {
              ...elementAllDataMock,
            },
          },
          dynamicData: {
            ['-1']: {
              ...currentPage.elements.dynamicData['-1'],
              children: [elementDynamicDataMock.id],
            },
            [elementDynamicDataMock.id]: elementDynamicDataMock,
          },
          staticData: {
            ['-1']: {
              ...currentPage.elements.staticData['-1'],
              children: [elementStaticDataMock.id],
            },
            [elementStaticDataMock.id]: {
              ...elementStaticDataMock,
            },
          },
        },
        selectedElements: [selectedElementMock],
      },
    },
  },
};

describe('ColumnPosition snapshots', () => {
  it('should render ColumnPosition', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ColumnPosition />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
