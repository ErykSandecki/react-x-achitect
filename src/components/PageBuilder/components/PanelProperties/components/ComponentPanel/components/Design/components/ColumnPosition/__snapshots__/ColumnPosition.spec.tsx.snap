// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnPosition snapshots should render ColumnPosition 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro SectionColumn SectionColumn--light"
    data-test-section=""
  >
    <div
      class="SectionColumn__labels SectionColumn__labels--light"
      style="width: calc(100% - 24px - 0px - 8px);"
    >
      <small
        class="css-q0izro SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Position
      </small>
    </div>
    <div
      class="SectionColumn__wrapper SectionColumn__wrapper--light"
    >
      <div
        class="SectionColumn__content SectionColumn__content--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <div
          class="css-q0izro TextField TextField--light"
          data-test-text-field="x"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="x"
          >
            <small
              class="css-q0izro Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              X
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="x"
            maxlength="6"
            type="number"
            value="0"
          />
        </div>
        <div
          class="css-q0izro TextField TextField--light"
          data-test-text-field="y"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="y"
          >
            <small
              class="css-q0izro Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Y
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="y"
            maxlength="6"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="SectionColumn__buttons SectionColumn__buttons--light"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ColumnPosition snapshots should render when values are mixed 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro SectionColumn SectionColumn--light"
    data-test-section=""
  >
    <div
      class="SectionColumn__labels SectionColumn__labels--light"
      style="width: calc(100% - 24px - 0px - 8px);"
    >
      <small
        class="css-q0izro SectionColumn__label SectionColumn__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Position
      </small>
    </div>
    <div
      class="SectionColumn__wrapper SectionColumn__wrapper--light"
    >
      <div
        class="SectionColumn__content SectionColumn__content--light"
        style="width: calc(100% - 24px - 0px - 8px);"
      >
        <div
          class="css-q0izro TextField TextField--light"
          data-test-text-field="x"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="x"
          >
            <small
              class="css-q0izro Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              X
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="x"
            maxlength="6"
            type="text"
            value="Mixed"
          />
        </div>
        <div
          class="css-q0izro TextField TextField--light"
          data-test-text-field="y"
        >
          <div
            class="ScrubbableInput ScrubbableInput--light"
            data-test-scrubbable-input="y"
          >
            <small
              class="css-q0izro Typography Typography--light Typography--neutral2 Typography--neutral2--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
              data-test-text=""
              style="text-align: inherit;"
            >
              Y
            </small>
          </div>
          <input
            class="TextField__input TextField__input--light"
            data-test-text-field-input="y"
            maxlength="6"
            type="text"
            value="Mixed"
          />
        </div>
      </div>
      <div
        class="SectionColumn__buttons SectionColumn__buttons--light"
      />
    </div>
  </div>
</DocumentFragment>
`;
